src/core/Block.ts

```diff
@@ -0,0 +1,255 @@
+import Handlebars from 'handlebars';
+import { v4 as uuidv4 } from 'uuid';
+
+import { EventBus } from './eventBus';
+
+type BlockMeta<P = any> = {
+  props: P;
+};
+
+type Events = Values<typeof Block.EVENTS>;
+
+export class Block<P = any> {
```

Comment:
Можно определить так
```suggestion
export class Block<P extends {
  events?: Record<string, () => void>
} = {}> {
```
Тогда не придется this.props приводить к any, чтобы получить events